packages/code-studio/src/main/SessionUtils.ts

```diff
@@ -83,15 +92,15 @@ export function createCoreClient(): CoreClient {
   return new dh.CoreClient(websocketUrl);
 }
 
-export async function requestParentLoginOptions(): Promise<LoginOptions> {
+export async function requestParentLoginOptions(): Promise<LoginOptionsResponse> {
   if (window.opener == null) {
     throw new Error('window.opener is null, unable to send auth request.');
   }
   return new Promise(resolve => {
     const listener = (
       event: MessageEvent<{
         message: string;
-        payload: LoginOptions;
+        payload: LoginOptionsResponse;
```

Comment:
I'm not a big fan of overloading the `loginOptions` directly... and then passing those wherever `loginOptions` are used or wherever we need this info. They're not really related...

Could we instead have another function `requestSessionDetailsFromParent` and have the parent respond to that instead?